Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -103,28 +97,39 @@ void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& pr
         return;
     }
 
-    if (!m_cookieJar) {
-        promise->reject(SecurityError);
-        return;
-    }
-
-    auto& document = *downcast<Document>(context);
-    auto completionHandler = [promise = WTFMove(promise)] (std::optional<Vector<Cookie>>&& cookies) {
-        if (!cookies) {
-            promise->reject(TypeError);
+    auto getCookies = [options = WTFMove(options), promise = WTFMove(promise)](ScriptExecutionContext& context) {
+        auto& document = downcast<Document>(context);
+        auto* page = document.page();
+        if (!page) {
+            promise->reject(SecurityError);
             return;
         }
 
-        auto& cookiesVector = *cookies;
-        if (cookiesVector.isEmpty()) {
-            promise->resolveWithJSValue(JSC::jsNull());
-            return;
-        }
+        auto& cookieJar = page->cookieJar();
+        auto completionHandler = [promise = WTFMove(promise)] (std::optional<Vector<Cookie>>&& cookies) {
+            if (!cookies) {
+                promise->reject(TypeError);
+                return;
+            }
+
+            auto& cookiesVector = *cookies;
+            if (cookiesVector.isEmpty()) {
+                promise->resolveWithJSValue(JSC::jsNull());
+                return;
+            }
 
-        promise->resolve<IDLDictionary<CookieListItem>>(CookieListItem(WTFMove(cookiesVector[0])));
+            promise->resolve<IDLDictionary<CookieListItem>>(CookieListItem(WTFMove(cookiesVector[0])));
+        };
+
+        cookieJar.getCookiesAsync(document, document.url(), options, WTFMove(completionHandler));
```

Comment:
I think using the document should work for service workers given document is constructed with the script URL and should have all the parameters of the service worker.
Maybe we will uncover differences and might have to change Page::setupForRemoteWorker if so.